### DISCUSSION TITLE
Debug iOS build Error in Flutter Workflow Editor

### DISCUSSION THREAD

**Discussion by mir-smarts1** [codemagic-ci-cd/codemagic-docs#1890] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [x] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I am on Windows. And I have tried to build a flutter debug iOS app in Codemagic. I am getting below error and no idea where it is coming:

Xcode build done.                                           23.6s
Failed to build iOS app
Lexical or Preprocessor Issue (Xcode): 'GoogleUtilities/GULURLSessionDataResponse.h' file not found
/Users/builder/clone/ios/Pods/GoogleDataTransport/GoogleDataTransport/GDTCCTLibrary/GDTCCTUploadOperation.m:36:8

Encountered error while building for device.

Build failed :|
Failed to build for iOS

### Expected results

successful worflow

### Actual results

I have not worked in iOS side of flutter app a lot. And I can not figure out the primary cause of the problem and how to figure out to fix it only in Codemagic Workflow Editor since I have no access to macOS

### Build id (optional)

64b7835cefa6cd8ba57590d3

**Comment by himesh-cm** [comment]
Hi @mir-smarts1, Can you check this [SO](https://stackoverflow.com/questions/66809424/googleutilities-gulurlsessiondataresponse-h-file-not-found) link if it helps, it is going to be tough to debug this without a mac machine.

**Reply by JasurbekErgashev** [reply, marked as answer]
Hey hello, thank you for your reply. Yeah, I have checked the link you mentioned. No luck. However, I am able to publish a new version to App Store successfully on a Windows machine. Basically, I deleted Podfile.lock file which was in version control. And I have a question. Is this file auto-generated while building, do we need to include this file in git and why, and what is the primary and main duty of this file since I have not worked with the iOS side of Flutter on a Mac device, I am just curious about these questions.
Thanks

**Reply by himesh-cm** [reply]
Yes, you should add the `Podfile.lock` to your git. This file ensures that everyone who clones the repository will have the exact same versions of the dependencies that you are using.

While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!
